### PR TITLE
Customer Match User List Action

### PR DESCRIPTION
**Pull request by maryamsharif**
[SDD](https://docs.google.com/document/d/12kh7-NLdWmAfKE0y12-JnGd_3wDs1a1LUUhpQnlPSxI/edit?pli=1)
[Testing Document
](https://docs.google.com/document/d/1pqFewqqqx-xa1yLzRtFuDweVGTr-QJj50z2jQf6B2SI/edit)
## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/google-enhanced-conversions/functions.ts [hunk 1]
```diff
@@ -185,3 +192,387 @@ export const commonHashedEmailValidation = (email: string): string => {
...
+  const json = {
+    query: `SELECT user_list.id, user_list.name FROM user_list`
+  }
+
+  try {
+    const response: ModifiedResponse<UserListResponse> = await request(
+      `https://googleads.googleapis.com/${API_VERSION}/customers/${settings.customerId}/googleAds:search`,
+      {
+        method: 'post',
+        headers: {
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/functions.ts:206]: Could we pull out the auth headers here and elsewhere into the top level `extendRequest`?

Comment (maryamsharif) [reply packages/destination-actions/src/destinations/google-enhanced-conversions/functions.ts:206]: I tried it this way before but unfortunately it doesn't work :/

File: packages/destination-actions/src/destinations/google-enhanced-conversions/functions.ts [hunk 2]
```diff
@@ -185,3 +192,387 @@ export const commonHashedEmailValidation = (email: string): string => {
...
+  const emailParts = normalizedEmail.split('@')
+  if (emailParts.length > 1 && emailParts[1].match(googleDomain)) {
+    emailParts[0] = emailParts[0].replace('.', '')
+    normalizedEmail = `${emailParts[0]}@${emailParts[1]}`
+  }
+
+  return sha256SmartHash(normalizedEmail)
+}
+
+function formatToE164(phoneNumber: string, defaultCountryCode: string): string {
```
Comment (nick-Ag) [inline packages/destination-actions/src/destinations/google-enhanced-conversions/functions.ts:370]: Could we maybe add a quick comment here on what E164 is?

File: packages/browser-destinations/webpack.config.js [hunk 3]
```diff
@@ -104,7 +104,8 @@ const unobfuscatedOutput = {
     mainFields: ['exports', 'module', 'browser', 'main'],
     extensions: ['.ts', '.js'],
     fallback: {
-      vm: require.resolve('vm-browserify')
+      vm: require.resolve('vm-browserify'),
+      crypto: false
```
Comment (nick-Ag) [inline packages/browser-destinations/webpack.config.js:108]: why was this change needed?

Comment (maryamsharif) [reply packages/browser-destinations/webpack.config.js:108]: I was getting build failure:

`Module not found: Error: Can't resolve 'crypto' in '/Users/maryamsharif/dev/src/github.com/segmentio/action-destinations/packages/core/dist/esm'`

so the change is for it to ignore the `crypto` module.

**Review by pooyaj — APPROVED** [review]

**Review by nick-Ag — APPROVED** [review]
